R/tm_gt_tbl_summary.R

```diff
@@ -0,0 +1,339 @@
+#' @param missing_text String indicating text shown on missing row.
+#' @param missing_stat statistic to show on missing row. Default is `"{N_miss}"`.
+#' Possible values are `N_miss`, `N_obs`, `N_nonmiss`, `p_miss`, `p_nonmiss`.
+#' @inheritSection gtsummary::tbl_summary statistic argument
+#' @inheritSection gtsummary::tbl_summary digits argument
+#' @inheritSection gtsummary::tbl_summary type and value arguments
+#' @section Decorating Module:
+#'
+#' This module generates the following objects, which can be modified in place using decorators:
+#' - `table` (`gtsummary` - output of [`crane::tbl_roche_summary()`])
+#'
+#' A Decorator is applied to the specific output using a named list of `teal_transform_module` objects.
+#' The name of this list corresponds to the name of the output to which the decorator is applied.
+#' See code snippet below:
+#'
+#' ```
+#' tm_gt_tbl_summary(
+#'    ..., # arguments for module
+#'    decorators = list(
+#'      table = teal_transform_module(...) # applied to the `table` output
+#'    )
+#' )
+#' ```
+#' For additional details and examples of decorators, refer to the vignette
+#' `vignette("decorate-module-output", package = "teal.modules.general")`.
+#'
+#' To learn more please refer to the vignette
+#' `vignette("transform-module-output", package = "teal")` or the [`teal::teal_transform_module()`] documentation.
+#'
+#' @inheritSection teal::example_module Reporting
+#' @export
+#' @importFrom methods is
+#' @examples
+#' data <- within(teal_data(), {
+#'   ADSL <- teal.data::rADSL
+#' })
+#' join_keys(data) <- default_cdisc_join_keys[names(data)]
+#' app <- init(
+#'   data = data,
+#'   modules = modules(
+#'     tm_gt_tbl_summary(
+#'       by = data_extract_spec(
+#'         dataname = "ADSL",
+#'         select = select_spec(
+#'           choices = c("SEX", "COUNTRY", "SITEID", "ACTARM"),
+#'           selected = "SEX",
+#'           multiple = FALSE
+#'         )
+#'       ),
+#'       include = data_extract_spec(
+#'         dataname = "ADSL",
+#'         select = select_spec(
+#'           choices = c("SITEID", "COUNTRY", "ACTARM"),
+#'           selected = "SITEID",
+#'           multiple = TRUE,
+#'           fixed = FALSE
+#'         )
+#'       )
+#'     )
+#'   )
+#' )
+#' if (interactive()) {
+#'   shinyApp(app$ui, app$server)
+#' }
+tm_gt_tbl_summary <- function(
+  label = "Table summary",
+  by = NULL,
+  col_label = NULL,
+  statistic = list(
+    gtsummary::all_continuous() ~ c("{mean} ({sd})", "{median}", "{min} - {max}"),
+    gtsummary::all_categorical() ~ "{n} ({p}%)"
+  ),
+  digits = NULL,
+  type = NULL,
+  value = NULL,
+  missing_text = "<Missing>",
+  missing_stat = "{N_nonmiss}",
+  sort = gtsummary::all_categorical(FALSE) ~ "alphanumeric",
+  include = tidyselect::everything(),
+  transformators = list(),
+  decorators = list()
+) {
+  message("Initializing tm_gt_tbl_summary")
+  checkmate::assert_string(label)
+  if (inherits(by, "data_extract_spec")) {
```

Comment:
If you want to use `tidyselect` and assert on this argument, then you need to handle this in a specific way. See here:
https://github.com/insightsengineering/teal.transform/blob/621603df655f4c051de612f1c35a66cf44a81756/R/0-picks.R#L427